activerecord/test/cases/encryption/encryptable_record_test.rb

```diff
@@ -134,6 +134,33 @@ class ActiveRecord::Encryption::EncryptableRecordTest < ActiveRecord::Encryption
     assert_equal "", book.name
   end
 
+  test "ciphertext length varies based on compression settings" do
+    # THRESHOLD_TO_JUSTIFY_COMPRESSION is 140 bytes
+    plaintext_under_threshold = "_" * 130
+    plaintext_over_threshold = "_" * 300
+
+    # Explicitly set the default
+    ActiveRecord::Encryption.config.compress = true
```

Comment:
You would need the attribute `compress` here, so that chignons in one test won't affect the whole suite.

https://github.com/rails/rails/blob/abfd2cc7a6eab91c558a30f7376c1b994055b2f4/activerecord/test/cases/encryption/helper.rb#L146-L147

In any case, I'd change the approach to make this an attribute-level setting as suggested and test it as as such.